### PR TITLE
TST: use manylinux2010 docker instead of ubuntu

### PR DESCRIPTION
**Pull request by mattip**
xref MacPython/numpy-wheels#73

If we already are using a docker to test 32-bit linux, let's use the manylinux2010 one.

### CODE REVIEW

**Comment by mattip** [comment]
This is now consistent with what I see locally when I use the docker to test. The failure fo `sin` is due to very large numbers escaping the AVX functions, and glibc's implementation is not consistent across platforms.

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
LGTM, I'll let you decide when to put it in due to the ongoing test fix.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -28,34 +28,32 @@ stages:
...
-            python3 -m pip install . && \
-            F77=gfortran-5 F90=gfortran-5 \
-            CFLAGS=-UNDEBUG python3 runtests.py -n --debug-info --mode=full -- -rsx --junitxml=junit/test-results.xml && \
+            python3 -m pip install -r test_requirements.txt && \
+            echo CFLAGS \$CFLAGS && \
+            python3 -m pip install -v . && \
+            python3 runtests.py -n --debug-info --mode=full -- -rsx --junitxml=junit/test-results.xml && \
             python3 tools/openblas_support.py --check_version $(OpenBLAS_version)"
-      displayName: 'Run 32-bit Ubuntu Docker Build / Tests'
+      displayName: 'Run 32-bit multilinux2010 Docker Build / Tests'
```
Comment (charris) [inline azure-pipelines.yml]: 'multilinux' <- 'manylinux'

I've made that same mistake so many times the name should probably be changed :)

Comment (mattip) [reply azure-pipelines.yml]: fixed

**Comment by mattip** [comment]
rebased on top of gh-15598

**Pull request merged**